system/logaccess/ElasticStack/ElasticStackLogAccess.hpp

```diff
@@ -0,0 +1,118 @@
+    StringBuffer m_componentsIndexSearchPattern;
+
+    StringBuffer m_audienceSearchColName;
+    StringBuffer m_audienceIndexSearchPattern;
+
+    StringBuffer m_classSearchColName;
+    StringBuffer m_classIndexSearchPattern;
+
+    StringBuffer m_defaultDocType; //default doc type to query
+
+    elasticlient::Client m_esClient;
+    StringBuffer m_esConnectionStr;
+
+public:
+
+#ifdef _CONTAINERIZED
+    //In containerized world, most likely Elastic Search host is their default k8s hostname
+    static constexpr const char * DEFAULT_ES_HOST = "elasticsearch-master";
+#else
+    //In baremetal, localhost is good guess as any
+    static constexpr const char * DEFAULT_ES_HOST = "localhost";
+#endif
+
+    static constexpr const char * DEFAULT_ES_PROTOCOL = "http";
+    static constexpr const char * DEFAULT_ES_DOC_TYPE = "_doc";
+    static constexpr const char * DEFAULT_ES_PORT = "9200";
+
+    static constexpr int DEFAULT_ES_DOC_LIMIT = 100;
+    static constexpr int DEFAULT_ES_DOC_START = 0;
+
+    static constexpr const char * DEFAULT_TS_NAME = "@timestamp";
+    static constexpr const char * DEFAULT_INDEX_PATTERN = "filebeat*";
+    static constexpr const char * DEFAULT_LOG_COLUMN_NAME = "message";
+
+    static constexpr const char * DEFAULT_HPCC_LOG_SEQ_COL = "hpcc.log.sequence";
+    static constexpr const char * DEFAULT_HPCC_LOG_TIMESTAMP_COL = "hpcc.log.timestamp";
+    static constexpr const char * DEFAULT_HPCC_LOG_MESS_COL = "hpcc.log.message";
+    static constexpr const char * DEFAULT_HPCC_LOG_AUD_COL = "hpcc.log.audience";
+    static constexpr const char * DEFAULT_HPCC_LOG_PROCID_COL = "hpcc.log.procid";
+    static constexpr const char * DEFAULT_HPCC_LOG_THREADID_COL = "hpcc.log.threadid";
+    static constexpr const char * DEFAULT_HPCC_LOG_TYPE_COL = "hpcc.log.class";
+    static constexpr const char * DEFAULT_HPCC_LOG_MESSAGE_COL = "hpcc.log.message";
+    static constexpr const char * DEFAULT_HPCC_LOG_JOBID_COL = "hpcc.log.jobid";
+    static constexpr const char * DEFAULT_HPCC_LOG_COMPONENT_COL = "kubernetes.container.name";
+
+    static constexpr const char * LOGMAP_INDEXPATTERN_PATH = "logmap/global/@storename";
+    static constexpr const char * LOGMAP_COLUMNNAME_PATH = "logmap/global/@searchcolumn";
+    static constexpr const char * LOGMAP_TIMESTAMPCOL_PATH = "logmap/global/@timestampcolumn";
+
+    ElasticStackLogAccess(const std::vector<std::string> &hostUrlList, IPropertyTree & logAccessPluginConfig);
+    virtual ~ElasticStackLogAccess() override = default;
+
+    virtual bool fetchLog(const LogAccessConditions & options, StringBuffer & returnbuf, LogAccessLogFormat format) override;
+    cpr::Response performESQuery(const LogAccessConditions & options);
+
+    const IPropertyTree * getESStatus();
+    const IPropertyTree * getIndexSearchStatus(const char * indexpattern);
+    const IPropertyTree * getTimestampTypeFormat(const char * indexpattern, const char * fieldname);
```

Comment:
trivial: my previous const comment, actually I meant the methods could/should be const not the return type, but perhaps both applicable

Reply:
Can't declare these methods const, they all call a non-const method on the m_esClient member